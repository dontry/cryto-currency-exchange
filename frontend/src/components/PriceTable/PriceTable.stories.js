import React from "react";
import { storiesOf } from "@storybook/react";
import PriceTable from "./index";
import prices from "../../assets/price.json";

storiesOf("PriceTable", module).add("with prices", () => (
  <PriceTable prices={prices} />
));
