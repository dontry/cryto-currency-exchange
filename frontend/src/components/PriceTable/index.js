import React from "react";
import PropTypes from "prop-types";
import {
  StyledTable,
  StyledTableHead,
  StyledTableRow,
  StyledTableHeadCell,
  StyledTableBody,
  StyledTableCell
} from "./style";
import { formatDate, formatTime } from "../../utils/index";

const head = ["Currency", "Date", "Time", "Price"];

const PriceTable = ({ prices = [], onClick }) => {
  const handleClick = index => () => {
    onClick(index);
  };
  return (
    <StyledTable>
      <StyledTableHead>
        <StyledTableRow>
          {head.map((title, index) => (
            <StyledTableHeadCell
              key={title}
              textAlign={index === head.length - 1 ? "right" : "center"}
            >
              {title}
            </StyledTableHeadCell>
          ))}
        </StyledTableRow>
      </StyledTableHead>
      <StyledTableBody>
        {prices.map((p, index) => {
          const { currency, date, time, price } = p;
          return (
            <StyledTableRow
              className="highlight"
              onClick={handleClick(index)}
              data-testid="table-row"
              key={`${currency}-${date}-${time}`}
            >
              <StyledTableCell>{currency}</StyledTableCell>
              <StyledTableCell>{formatDate(date)}</StyledTableCell>
              <StyledTableCell>{formatTime(time)}</StyledTableCell>
              <StyledTableCell textAlign="right">${price}</StyledTableCell>
            </StyledTableRow>
          );
        })}
      </StyledTableBody>
    </StyledTable>
  );
};

PriceTable.propTypes = {
  prices: PropTypes.arrayOf(
    PropTypes.shape({
      currency: PropTypes.string,
      date: PropTypes.string,
      quotes: PropTypes.arrayOf(
        PropTypes.shape({
          time: PropTypes.string,
          price: PropTypes.string
        })
      )
    })
  ),
  onClick: PropTypes.func
};

export default PriceTable;
