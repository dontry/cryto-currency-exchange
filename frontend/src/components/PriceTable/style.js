import styled from "styled-components";

export const StyledTable = styled.table`
  width: 100%;
  display: table;
  border-spacing: 0;
  border-collapse: collapse;
  border: 1px solid rgba(224, 224, 224, 1);
`;

export const StyledTableHead = styled.thead`
  display: table-header-group;
`;

export const StyledTableRow = styled.tr`
  color: inherit;
  display: table-row;
  outline: none;
  vertical-align: middle;
`;

export const StyledTableHeadCell = styled.th`
  display: table-cell;
  color: rgba(0, 0, 0, 0.54);
  font-weight: 500;
  line-height: 1.3125rem;
  padding: 14px 40px 14px 16px;
  font-size: 0.875rem;
  text-align: ${props => props.textAlign || "center"};
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  &:last-child {
    padding-right 40px;
  }
`;

export const StyledTableBody = styled.tbody`
  display: table-row-group;
`;

export const StyledTableCell = styled.td`
  display: table-cell;
  padding: 14px 40px 14px 16px;
  font-size: 0.875rem;
  text-align: ${props => props.textAlign || "center"};
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1.43;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  letter-spacing: 0.01071em;
  vertical-align: inherit;

  &:last-child {
    padding-right 40px;
  }
`;
