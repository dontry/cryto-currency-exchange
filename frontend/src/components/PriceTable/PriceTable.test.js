import React from "react";
import PriceTable from "./PriceTable";
import { render } from "@testing-library/react";
import prices from "../../../assets/price.json";

describe("PriceTable", () => {
  it("should render table with currency data", () => {
    const { container, queryAllByTestId, queryAllByText } = render(
      <PriceTable prices={[prices[0]]} />
    );
    expect(queryAllByTestId("table-row").length).toBe(prices[0].quotes.length);
    expect(queryAllByText(prices[0].currency).length).toBe(
      prices[0].quotes.length
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
