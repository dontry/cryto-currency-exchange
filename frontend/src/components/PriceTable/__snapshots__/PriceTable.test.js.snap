// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceTable should render table with currency data 1`] = `
<table
  class="sc-bdVaJa dRchuH"
>
  <thead
    class="sc-bwzfXH eLxnQL"
  >
    <tr
      class="sc-htpNat ngmkp"
    >
      <th
        class="sc-bxivhb ccwLpn"
      >
        Currency
      </th>
      <th
        class="sc-bxivhb ccwLpn"
      >
        Date
      </th>
      <th
        class="sc-bxivhb ccwLpn"
      >
        Time
      </th>
      <th
        class="sc-bxivhb ccwLpn"
      >
        Price
      </th>
    </tr>
  </thead>
  <tbody
    class="sc-ifAKCX eRQhwu"
  >
    <tr
      class="sc-htpNat ngmkp"
      data-testid="table-row"
    >
      <td
        class="sc-EHOje kxMuyp"
      >
        BTC
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        20180507
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        0915
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        34.98
      </td>
    </tr>
    <tr
      class="sc-htpNat ngmkp"
      data-testid="table-row"
    >
      <td
        class="sc-EHOje kxMuyp"
      >
        BTC
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        20180507
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        1045
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        36.13
      </td>
    </tr>
    <tr
      class="sc-htpNat ngmkp"
      data-testid="table-row"
    >
      <td
        class="sc-EHOje kxMuyp"
      >
        BTC
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        20180507
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        1230
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        37.01
      </td>
    </tr>
    <tr
      class="sc-htpNat ngmkp"
      data-testid="table-row"
    >
      <td
        class="sc-EHOje kxMuyp"
      >
        BTC
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        20180507
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        1400
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        35.98
      </td>
    </tr>
    <tr
      class="sc-htpNat ngmkp"
      data-testid="table-row"
    >
      <td
        class="sc-EHOje kxMuyp"
      >
        BTC
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        20180507
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        1530
      </td>
      <td
        class="sc-EHOje kxMuyp"
      >
        33.56
      </td>
    </tr>
  </tbody>
</table>
`;
